### PR TITLE
oprf: Update deterministic key generation.

### LINKED ISSUE #225: oprf: Update deterministic key derivation
https://github.com/cfrg/draft-irtf-cfrg-voprf/pull/247

### PR DESCRIPTION
**Pull request by armfazh**
Fixes #225

### CODE REVIEW

File: oprf/keys.go [hunk 1]
```diff
@@ -44,19 +47,24 @@ func (k *PrivateKey) Public() *PublicKey {
...
 
 // GenerateKey generates a pair of keys in accordance with the suite.
-func GenerateKey(id SuiteID) (*PrivateKey, error) {
+func GenerateKey(id SuiteID, rnd io.Reader) (*PrivateKey, error) {
 	suite, err := suiteFromID(id, BaseMode)
 	if err != nil {
 		return nil, err
 	}
-	return suite.generateKey(), nil
+	if rnd == nil {
```
Comment (wbl) [inline oprf/keys.go:55]: Why can't the caller pass in rand.Reader themselves?

**Review by wbl — APPROVED** [review]

File: oprf/keys.go (hunk 1 above)
Comment (armfazh) [reply oprf/keys.go:55]: now, it panics. please review changes.

Comment (wbl) [reply oprf/keys.go:55]: I think you might not even need the if and let the read function panic.

Comment (armfazh) [reply oprf/keys.go:55]: Here we are following the api pattern in go stdlib.
e.g. https://golang.org/pkg/crypto/ecdsa/#GenerateKey

**Pull request merged**